src/extension.ts

```diff
@@ -407,6 +408,53 @@ function installCustomRequestHandlers(context: VSCode.ExtensionContext) {
   languageClient.onRequest(protocol.ShowTaintVulnerabilityRequest.type, showAllLocations);
 }
 
+async function isFileIgnoredForFolder(workspaceFolderPath: string, gitCommand: string): Promise<boolean> {
+  const {sout, serr} = await new Promise<{ sout: string, serr: string }>((resolve, reject) => {
+    ChildProcess.exec(
+        gitCommand,
+        {cwd: workspaceFolderPath},
+        (error: Error & { code?: 0 | 1 | 128 }, stdout, stderr) => {
+          if (error && (error.code !== 0 && error.code !== 1)) {
+            logToSonarLintOutput('Error on git gitCommand: ' + error);
+            reject(error);
+            return;
+          }
+          resolve({sout: stdout, serr: stderr});
+        }
+    );
+  });
+
+  if (serr) {
+    throw new Error(serr);
+  }
+
+  return Promise.resolve(sout.length > 0);
+}
+
+async function isIgnoredByScm(fileUri: string): Promise<boolean> {
+  const uriRelative = VSCode.workspace.asRelativePath(fileUri, false);
+  const command = `git check-ignore ${uriRelative}`;
```

Comment:
So if you open a workspace folder, and then open a file that is outside of it, it will fail with this error:
```
[Debug - 18:01:27.490] Queuing analysis of file 'file:///home/julien/Prog/ignored.js'
Error on git gitCommand: Error: Command failed: git check-ignore /home/julien/Prog/ignored.js
fatal: /home/julien/Prog/ignored.js : '/home/julien/Prog/ignored.js' est hors du dépôt à '/home/julien/Prog/Projects/sonarlint/sonarlint-vscode'

[stderr] Exception in thread "SonarLint analysis" java.util.concurrent.CompletionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: Request sonarlint/isIgnoredByScm failed with message: Command failed: git check-ignore /home/julien/Prog/ignored.js
[stderr] fatal: /home/julien/Prog/ignored.js : '/home/julien/Prog/ignored.js' est hors du dépôt à '/home/julien/Prog/Projects/sonarlint/sonarlint-vscode'
[stderr] 	at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412)
[stderr] 	at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2044)
[stderr] 	at org.sonarsource.sonarlint.ls.AnalysisManager.lambda$analyze$5(AnalysisManager.java:307)
[stderr] 	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
[stderr] 	at org.sonarsource.sonarlint.ls.AnalysisManager.analyze(AnalysisManager.java:307)
[stderr] 	at org.sonarsource.sonarlint.ls.AnalysisManager.lambda$analyzeAsync$4(AnalysisManager.java:297)
[stderr] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[stderr] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[stderr] 	at java.base/java.lang.Thread.run(Thread.java:829)
```

Reply:
> est hors du dépôt à

means

> is outside repository at